pytroll_runner/__init__.py

```diff
@@ -47,12 +50,31 @@ def parse_args(args=None):
 def run_and_publish(config_file):
     """Run the command and publish the expected files."""
     command_to_call, subscriber_config, publisher_config = read_config(config_file)
+    preexisting_files = check_existing_files(publisher_config)
+
     with closing(create_publisher_from_dict_config(publisher_config["publisher_settings"])) as pub:
-        for _, mda in run_from_new_subscriber(command_to_call, subscriber_config):
-            message = generate_message_from_expected_files(publisher_config, mda)
+        for log_output, mda in run_from_new_subscriber(command_to_call, subscriber_config):
+            try:
+                message = generate_message_from_log_output(publisher_config, mda, log_output)
+            except KeyError:
+                message = generate_message_from_expected_files(publisher_config, mda, preexisting_files)
+                preexisting_files = check_existing_files(publisher_config)
             pub.send(message)
 
 
+def generate_message_from_log_output(publisher_config, mda, log_output):
+    """Generate message for the filenames present in the log output."""
+    new_files = re.findall(publisher_config["output_files_log_regex"], str(log_output))
```

Comment:
So if there's a regex pattern in the config, the output from the called script is parsed to catch the filenames? Handy.

Reply:
yes exactly.